### PR TITLE
Chore: (Docs) Updates to the CLI

### PR DESCRIPTION
**Pull request by jonniebigodes**
With this pull request, the CLI reference is updated to address the recent changes based on the removal of the binaries and changes to the scripts provided by Storybook upon installation. In addition to this, some additional references were also removed from the documentation as they were referencing scripts that are no longer in use.

@shilman, when you have a chance, can you take a look and let me know if there's anything else missing so that I can make the updates and we can merge it. Thanks in advance

### CODE REVIEW

File: docs/api/cli-options.md [hunk 1]
```diff
@@ -2,69 +2,73 @@
...
-Storybook comes with two CLI utilities: [`start-storybook`](#start-storybook) and [`build-storybook`](#build-storybook).
+The Storybook command line interface (CLI) is the main tool you use to build and develop Storybook.
 
-<div class="aside">
-
-Storybook collects completely anonymous data to help us improve user experience. Participation is optional, and you may [opt-out](../configure/telemetry.md#how-to-opt-out) if you'd not like to share any information.
+## API commands
 
-</div>
+All the following documentation is available in the CLI by running `storybook --help`.
```
Comment (kylegach) [inline docs/api/cli-options.md:9]:
```suggestion
All of the following documentation is available in the CLI by running `storybook --help`.
```

File: docs/configure/telemetry.md [hunk 2]
```diff
@@ -24,7 +24,7 @@ We collect general usage details, including command invocation, Storybook versio
 Specifically, we track the following information in our telemetry events:
 
 - Timestamp of the occurrence.
-- Command invoked (e.g., `init`, `upgrade`, `start-storybook`).
+- Command invoked (e.g., `init`, `upgrade`, `storybook`).
```
Comment (kylegach) [inline docs/configure/telemetry.md:27]: `init` and `upgrade` are commands of the `storybook` CLI, so I think `dev` would be the better analog here. I went ahead and added `build`, too.

```suggestion
- Command invoked (e.g., `init`, `upgrade`, `dev`, `build`).
```

File: docs/api/cli-options.md [hunk 3]
```diff
@@ -2,69 +2,73 @@
...
 
-## build-storybook
+### `build`
+
+Compiles your Storybook instance so it can be [deployed](../sharing/publish-storybook.md). Should be run from the root of your project.
 
-```plaintext
-Usage: build-storybook [options]
+```shell
+storybook dev
```
Comment (kylegach) [inline docs/api/cli-options.md:53]: Typo

```suggestion
storybook build
```

File: docs/api/cli-options.md [hunk 4]
```diff
@@ -2,69 +2,73 @@
 title: 'CLI options'
 ---
 
-Storybook comes with two CLI utilities: [`start-storybook`](#start-storybook) and [`build-storybook`](#build-storybook).
+The Storybook command line interface (CLI) is the main tool you use to build and develop Storybook.
 
-<div class="aside">
```
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply docs/api/cli-options.md:7]: Removed accidentally and added it back with the latest commit

**Pull request merged**